### PR TITLE
UI Changes + Download button

### PR DESCRIPTION
**Pull request by btorra-onshape**
- formatting the top bar
- 3rd party cookies
- dropdown feature
- help button
- assemblies show

### CODE REVIEW

File: web/js/index.js [hunk 1]
```diff
@@ -278,7 +278,44 @@ fetch(`/api/elements${window.location.search}`, { headers: { 'Accept': 'applicat
...
+            }
         }
     }).catch((err) => {
         displayError(`Error while requesting document elements: ${err}`);
     });
+
+const dropdown = document.getElementById('elem-selector');
+const selectedOptionText = document.getElementById('selected-option');
+
+dropdown.addEventListener('change', function() {
```
Comment (pbhide17) [inline web/js/index.js:295]: As far as I know, we don't need to write this function.
The select HTML element gives us the functionality that is implemented in this event listener.

File: web/js/index.js [hunk 2]
```diff
@@ -278,7 +278,44 @@ fetch(`/api/elements${window.location.search}`, { headers: { 'Accept': 'applicat
...
+    selectedOptionText.textContent = selectedValue;
+});
+
+document.getElementById('elem-selector').addEventListener('click', onClickHandler);
+document.getElementById('elem-selector').addEventListener('mousedown', onMouseDownHandler);
+
+function onMouseDownHandler(e){
+	var el = e.currentTarget;
+	
+    // when we click, don't make it choose an option
```
Comment (pbhide17) [inline web/js/index.js:306]: Shouldn't we allow choosing an option when the option is clicked?

File: web/js/index.js [hunk 3]
```diff
@@ -278,7 +278,44 @@ fetch(`/api/elements${window.location.search}`, { headers: { 'Accept': 'applicat
...
+}
+
+function onClickHandler(e) {
+ 	var el = e.currentTarget; 
+
+    if (el.getAttribute('size') == '1') {
+        el.setAttribute('size', '5');
+    }
+    else {
+        el.setAttribute('size', '1');
```
[the remark this answers is not in the document]
Comment (btorra-onshape) [reply web/js/index.js:319]: Right, working on trying to get that to work.

[the remark this answers is not in the document]
Comment (btorra-onshape) [reply web/js/index.js:319]: This is a temp solution at the moment

File: app.js [hunk 4]
```diff
@@ -57,12 +57,16 @@ app.use('/oauthSignin', (req, res) => {
...
-}, (req, res) => {     console.log("Redirecting to Onshape for authentication");/* redirected to Onshape for authentication */ });
+}, (req, res) => { /* redirected to Onshape for authentication */ });
 
 app.use('/oauthRedirect', passport.authenticate('onshape', { failureRedirect: '/grantDenied' }), (req, res) => {
-    res.redirect(`/?documentId=${req.session.state.docId}&workspaceId=${req.session.state.workId}&elementId=${req.session.state.elId}`);
+    if (req.session.state) {
+        res.redirect(`/?documentId=${req.session.state.docId}&workspaceId=${req.session.state.workId}&elementId=${req.session.state.elId}`);
+    }
+    else {
+        res.sendFile(path.join(__dirname, 'public', 'html', 'cookiesDisabled.html'));
```
Comment (PaulJPremakumar) [inline app.js:68]: Can you add a comment here as to what this path of code is? For example // display third party cookies disabled if unable to save req.session.state

File: public/css/index.css [hunk 5]
```diff
@@ -1,8 +1,49 @@
...
+    top: 20px;
+    right: 20px;
+}
+
+#selected-option {
+    margin-top: 0px;
+    font-family: 'Flama', sans-serif;
+    color: #64bc64;
+    font-size: x-large;
+}
```
Comment (PaulJPremakumar) [inline public/css/index.css:49]: Add a new line here. It is good practice to add a new line at the end of a file. For more information : https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

File: public/html/cookiesDisabled.html [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<!DOCTYPE html>
+<html>
+<body>
+  <div>
+    <strong> You have 3rd party cookies disabled. In order to use the glTF viewer, please enable 3rd party cookies. </strong>
```
Comment (PaulJPremakumar) [inline public/html/cookiesDisabled.html:5]: If Gaby has not reviewed the wording, please have her review it.

File: web/js/index.js [hunk 7]
```diff
@@ -278,7 +278,44 @@ fetch(`/api/elements${window.location.search}`, { headers: { 'Accept': 'applicat
...
+    }
+}
```
Comment (PaulJPremakumar) [inline web/js/index.js:321]: New line here as well.

File: web/js/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (pbhide17) [reply web/js/index.js]: Done.

File: web/js/index.js (hunk 2 above)
Comment (pbhide17) [reply web/js/index.js]: Fixed.

File: public/css/index.css (hunk 5 above)
Comment (pbhide17) [reply public/css/index.css]: Done.

File: web/js/index.js (hunk 7 above)
Comment (pbhide17) [reply web/js/index.js]: Done.

File: web/js/index.js (hunk 1 above)
Comment (pbhide17) [reply web/js/index.js]: Done.

**Comment by pbhide17** [comment]
I have fixed the styling per Gaby's guidelines, and fixed the dropdown issues Brett faced.
I have also added functionality to clear the canvas before loading a glTF. (In case there's an error, we won't be showing an outdated glTF)

Currently working on download functionality.
The ability to download glTF is done, I just need a design template from UI team for how to let the user choose between a .json glTF download and a .glb download.

**Comment by pbhide17** [comment]
Right now I have only added support for downloading .gltf files.
We can add a dialog that lets the user select which type of file they want, or we can keep the app simple and say that only .gltf downloads are allowed in this app.

File: app.js (hunk 4 above)
Comment (pbhide17) [reply app.js:69]: Done

**Review by PaulJPremakumar — APPROVED** [review]

**Pull request merged**